---
<h1 align='center'>Logistic Regression</h1>

---

#### we will evaluate a LogisticRegression model using all the available features.

#### The model is fit on the training dataset and evaluated on the test dataset.

#### The complete example is listed below.

In [18]:
import numpy as np
import pandas as pd
from sklearn.preprocessing import OrdinalEncoder
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score


# load dataset
# create function for load dataset
def load_dataset(filename):
    # load dataset as a pandas fataframe
    data=pd.read_csv(filename,header=None)
    
    # retrieve numpy  array
    dataset=data.values
    
    # Split into Input(X) and output(y) variables
    X=dataset[:,:-1]
    y=dataset[:,-1]
    
    # format all fileds as string
    X=X.astype(str)
    return X,y

# prepare input data
def prepare_inputs(X_train,X_test):
    oe=OrdinalEncoder()
    oe.fit(X_train)
    X_train_enc=oe.transform(X_train)
    X_test_enc=oe.transform(X_test)
    return X_train_enc,X_test_enc


# Prepare target
def prepare_targets(y_train,y_test):
    le=LabelEncoder()
    le.fit(y_train)
    y_train_enc=le.transform(y_train)
    y_test_enc=le.transform(y_test)
    return y_train_enc,y_test_enc

# load the dataset
X, y = load_dataset('data/breast-cancer.csv')
# split into train and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=1)
# prepare input data
X_train_enc, X_test_enc = prepare_inputs(X_train, X_test)
# prepare output data
y_train_enc, y_test_enc = prepare_targets(y_train, y_test)

# Fit the model
model=LogisticRegression(solver='lbfgs')
model.fit(X_train_enc,y_train_enc)

# Evaluate the model
yhat=model.predict(X_test_enc)

# evaluate predictions
accuracy=accuracy_score(y_test_enc,yhat)
print("Accuracy : %.2f"%(accuracy*100))


Accuracy : 72.09


#### Running the example prints the accuracy of the model on the training dataset.

#### Note: your specific results may vary given the stochastic nature of the learning algorithm. Try running the example a few times.

#### In this case, we can see that the model achieves a classification accuracy of about 72%.

#### We would prefer to use a subset of features that achieves a classification accuracy that is as good or better than this.